File: Java/Calc.java
import org.antlr.v4.runtime.*;
import org.antlr.v4.runtime.tree.*;
import java.io.BufferedReader;
import java.io.InputStreamReader;

public class Calc {
    public static void main(String[] args) throws Exception {
        EvalVisitor eval = new EvalVisitor();
        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));

        System.out.println("Calculadora ANTLR4 (escribe 'exit' para salir)");
        System.out.println("Modo actual: GRADOS");
        System.out.println("Comandos: mode=deg | mode=rad");

        while (true) {
            System.out.print("> ");
            String line = reader.readLine();
            if (line == null || line.equalsIgnoreCase("exit")) break;

            if (line.equalsIgnoreCase("mode=deg")) {
                eval.setUseDegrees(true);
                System.out.println("Modo cambiado a GRADOS");
                continue;
            }
            if (line.equalsIgnoreCase("mode=rad")) {
                eval.setUseDegrees(false);
                System.out.println("Modo cambiado a RADIANES");
                continue;
            }

            CharStream input = CharStreams.fromString(line + "\n");
            LabeledExprLexer lexer = new LabeledExprLexer(input);
            CommonTokenStream tokens = new CommonTokenStream(lexer);
            LabeledExprParser parser = new LabeledExprParser(tokens);
            ParseTree tree = parser.stat();
            System.out.println("Árbol sintáctico:");
            System.out.println(tree.toStringTree(parser));
            System.out.println("Resultado:");
            eval.visit(tree);
        }
    }
}

File: calc.py
import sys
from antlr4 import *
from LabeledExprLexer import LabeledExprLexer
from LabeledExprParser import LabeledExprParser
from EvalVisitor import EvalVisitor

def main():
    visitor = EvalVisitor()
    print("Calculadora ANTLR4 (Python)")
    print("Escribe 'exit' para salir")
    print("Modo actual: GRADOS")
    print("Comandos: mode=deg | mode=rad")

    while True:
        try:
            line = input("> ")
        except EOFError:
            break

        if line.strip().lower() == "exit":
            break

        if line.strip().lower() == "mode=deg":
            visitor.useDegrees = True
            print("Modo cambiado a GRADOS")
            continue

        if line.strip().lower() == "mode=rad":
            visitor.useDegrees = False
            print("Modo cambiado a RADIANES")
            continue

        input_stream = InputStream(line + "\n")
        lexer = LabeledExprLexer(input_stream)
        tokens = CommonTokenStream(lexer)
        parser = LabeledExprParser(tokens)
        tree = parser.stat()

        visitor.visit(tree)

if __name__ == '__main__':
    main()
